counsel.el

```diff
@@ -5096,6 +5100,44 @@ build environments.")
                      (or (buffer-file-name) default-directory)
                      ".git")))
 
+;; This is loosely based on the bash completion code
+(defun counsel--get-make-calls (root)
+  "Return a list of make calls for a given directory.
+
+We search the Makefile for a list of PHONY targets which are generally
+the top-level targets a make system provides."
+  (let ((default-directory root))
+    (mapcar
+     (lambda(target)
+       (format "cd %s; make %s %s" root counsel-compile-make-opts target))
+     (split-string
+      (shell-command-to-string
+       (concat "make -nqp |"
+               "grep -B 1 PHONY |"
+               "grep ':' |"
+               "cut -d ':' -f 1 |"
+               "sort"))
+      "\n"))))
+
+(defun counsel-compile-get-make-invocaton (&optional build-root)
```

Comment:
The name looks misleading.  How about `counsel-compile-get-build-command` or `counsel-compile-command`?